bot.py

```diff
@@ -712,6 +719,65 @@ async def clear_queue(self, message, _command_content):
         self._stop()
         await message.add_reaction(MusicBot.REACTION_EMOJI)
 
+    async def remove_song_from_deque(self, message, command_content):
+        """
+        Removes a specific song from the queue, based on its position therein
+        """
+        if len(self.media_deque) == 0:
+            await message.channel.send(":unamused: The queue is empty...")
+            return
+
+        arg = command_content.split(maxsplit=1)[0]
+
+        range_regex = re.compile(r"(\d+)-(\d+)")
+        range_regex_match = range_regex.match(arg)
+        if range_regex_match:
+            await self.remove_range_from_deque(message, range_regex_match)
+            return
+
+        if arg.lower() == "first":
+            song_index = 0
+        elif arg.lower() == "last":
+            song_index = -1
+        elif arg.isnumeric():
+            song_index = int(arg) - 1
+        else:
+            await message.channel.send(
+                f"{arg} isn't a valid song position/range :open_mouth:"
+            )
+            return
+
+        if song_index < -1 or song_index >= len(self.media_deque):
+            await message.channel.send(
+                "Pick something *in* the queue! :face_with_symbols_over_mouth:"
+            )
+            return
+
+        del self.media_deque[song_index]
+        await message.add_reaction(MusicBot.REACTION_EMOJI)
+
+    async def remove_range_from_deque(self, message, range_regex_match):
```

Comment:
How about calling this function with the actual lower and upper bounds of the regex? You can even use list expansion when calling:
`await self.remove_range_from_deque(message, *range_regex_match.groups())`

```suggestion
    async def remove_range_from_deque(self, message, range_lower, range_upper):
```

Reply:
Yes, that's better 👍